Gem/Nodes/Services/LatencyMeasureService.swift

```diff
@@ -0,0 +1,57 @@
+// Copyright (c). Gem Wallet. All rights reserved.
+
+import Foundation
+
+struct LatencyMeasureService {
+    enum Latency {
+        case fast(Double)
+        case normal(Double)
+        case slow(Double)
+
+        init(latency: Double) {
+            switch latency {
+            case ..<420:
+                self = .fast(latency)
+            case 420..<840:
+                self = .normal(latency)
+            default:
+                self = .slow(latency)
+            }
+        }
+
+        private var colorEmoji: String {
+            switch self {
+            case .fast: return "🟢"
+            case .normal: return "🟠"
+            case .slow: return "🔴"
+            }
+        }
+
+        var value: Int {
+            let value: Double
+            switch self {
+            case .fast(let double):
+                value = double
+            case .normal(let double):
+                value = double
+            case .slow(let double):
+                value = double
+            }
+            return Int(value)
+        }
+
+        var formattedValue: String {
+            let ms = Localized.Common.ms
+            return "\(value) \(ms) \(colorEmoji)"
+        }
+    }
+
+    static func measure<T>(for asyncFunction: @escaping () async throws -> T?) async throws -> (latency: Latency, value: T?) {
```

Comment:
Did you consider extending SwiftHttpClient to provide request method with latency instead?

Reply:
Current implementation of getting ping can be appaielble as "adapter pattern" to any async await request/service/whatever we have. 

If we consider to using it in SwiftHttpClient by extending the functionality of request/response with several methods, we also have to extend our existing api services, if I get it correctly orr import SwiftHHTPClient in our viewmodels, modules - which is a one more external dependecy with a limit usage ( because we need a ping only in few places in gem target )